### ISSUE TITLE
Loading panel is hided too soon when using progressiveLoad in "load" mode

### ISSUE THREAD

**Issue by panofr** [olifolkerd/tabulator#3854]
**Describe the bug**
When using Ajax progressiveLoad='load', the loading panel is hided just after treating the first ajax answer.
Events "dataProcessed" and "dataProcessing" are dispatched only once (after first ajaxResponse).
Event "dataLoaded" is never dispatched.

**Tabulator Info**
Tabulator 5.3.0

**Working Example**
[dataLoaderLoading](https://codesandbox.io/s/tabulator-bug-event-progressive-load-8gytqr?file=/src/index.js)

**To Reproduce**
A step by step guide to recreate the issue in your JS Fiddle or Codepen:
1. Go to the sample page
2. The 'loading' panel disappear after few miliseconds.
3. Wait some times (less than 10 sec)
4. Open 'console' in bottom right, events "dataLoaded"/"dataProcessing"/"dataProcessed" were not dispacthed (or only once at start)

**Expected behavior**
In progressiveLoad="scroll" I understand why it's mandatory to instantly hide 'loading' panel, but the behaviour should be the opposite in 'load' mode, no?
If I understand correctly tabulator concepts, 'dataLoaded' events should be dispatched for each 'dataLoading' event and 'dataProcessed' should be dispatched only when all data have been rendered.

**Screenshots**
![bug_tabulator](https://user-images.githubusercontent.com/159509/178499022-443991b6-f6a0-4efc-a88f-83db697bc69f.png)

**Desktop (please complete the following information):**
 - OS: Windows 10 / Windows 11
 - Browser : Chrome 
 - Version 103.0.5060.114

**Comment by olifolkerd** [comment]
Hey @panofr 

The loader is intentionally hidden as soon as the first page is loaded, if it stayed on until all the data loaded then it wouldnt be progressive loading, you might as well just load the data in one chunk, the purpose of the progressive loading is to allow the user to use the table while the data is being loaded which is why the loader is hidden as soon as there is data in the table.

The `dataProcessing` events are not fired because the additional pages of data are loaded into the table by adding the rows individually rather than replacing the table data which is what triggers the `dataProcessing` events.

If you could explain your usage case i may be able to suggest the correct event for your needs.

Cheers

Oli :)

**Comment by panofr** [comment]
Hey @olifolkerd 

thanks for your fast reply. I understand better now why Tabulator is using this behaviour.

My usage is quite specific:

I have created a C++ service that broadcasts audio playlists (songs/podcasts...). 
I want to display the audio playlist in one table with vertical 'infinite' scrolling (no real paging). 
I can get each playlist by AJAX requests (by chunk of 50 rows at once, because it's too slow to 1000+ rows on one call).
Each row is quite simple : broadcast time, duration, title.

**Tabulator with progressiveLoad works really well for this.**

But I also want to 'update' the table each time a song is finished, or even added, edited, removed (I just know that something has changed, but I can't detect which row(s) exactly have been changed).
And it's my problem, because  I have to call 'setData' that get and redraw the full table, so we can see the "scrollbar expending" slowly and I lost the current vertical scroll position (another ticket is open for this point: https://github.com/olifolkerd/tabulator/issues/3372)
So I wanted to keep the table in 'loading' state, until the reception of a 'final' event to trigger the scroll function and get back to my position.

Maybe the best way in this case is to get the full playlist 'by myself' and call 'updateData' with the new version directly without using progressiveLoad option?

Notes: Each 'row' has an unique 'id' but I didn't manage to make the 'updateData' work in this case.

I'm sorry my english and my explanation skills are not very good :)

Best regards,

Lionel

**Comment by olifolkerd** [comment]
Hey @panofr 

You should use the 'replaceData' function when you want to update the table but keep the data in place.

Though as with all the other options this wont play well with progessive loading, the reason for this is when you call udpate/replace the data you might be updating may not have loaded into the table yet, so to be sure Tabulator reloads it all.

You wouldnt be able to use both approaches simultaneously.

In this scenario it may be better for you to chunk the data yourself using the `replaceData` function to load in the chunks into the table individually and also to load in any updates.

I hope that helps,

Cheers

OIi :)
